lstm_stock_prediction/lstm_stock_prediction.cpp

```diff
@@ -93,38 +93,37 @@ void SaveResults(const string filename,
                  data::MinMaxScaler& scale,
                  const arma::cube& testX)
 {
-  arma::mat flatDataAndPreds = testX.slice(testX.n_slices - 1);
-
-  // The prediction results are the (high, low) for the next day and come from
-  // the last slice from the prediction.
-  flatDataAndPreds.rows(flatDataAndPreds.n_rows - 2,
-                        flatDataAndPreds.n_rows - 1) =
-      predictions.slice(predictions.n_slices - 1);
-
-  scale.InverseTransform(flatDataAndPreds, flatDataAndPreds);
-
-  // We need to remove the last column because it was not used for training
-  // (there is no next day to predict).
-  flatDataAndPreds.shed_col(flatDataAndPreds.n_cols - 1);
-
-  // Save the data to file. The last columns are the predictions; the preceding
-  // columns are the data used to generate those predictions.
-  data::Save(filename, flatDataAndPreds);
-
-  // Print the output to screen.
-  // NOTE: we do not have the last data point in the input for the prediction
-  // because we did not use it for the training, therefore the prediction result
-  // will be for the day before. In your own application you may of course load
-  // any dataset for prediction.
-  cout << "The predicted Google stock (high, low) for the last day is: "
-       << endl;
-  cout << "  ("
-       << flatDataAndPreds(flatDataAndPreds.n_rows - 2,
-                           flatDataAndPreds.n_cols - 1)
-       << ", ";
-  cout << flatDataAndPreds(flatDataAndPreds.n_rows - 1,
-                           flatDataAndPreds.n_cols - 1)
-       << ")" << endl;
+  arma::mat flatDataAndPreds = testX.slice(testX.n_slices - 1);
+  scale.InverseTransform(flatDataAndPreds, flatDataAndPreds);
+
+  //The prediction results are the (high, low) for the next day and comming from 
+  //the last slice from the prediction.
+  arma::mat temp = predictions.slice(predictions.n_slices - 1);
+
+  //NOTE: We add 3 extra rows here in order to recreate the input data 

```

Comment:
```suggestion
  // NOTE: We add 3 extra rows here in order to recreate the input data
```
Thanks for this comment.  If I had seen it in my initial review I would have had a lot fewer questions.  I must have clicked on the link wrong or something. :(